### PR TITLE
[WIP] Deprecate solve_linear_system for linsolve

### PR DESCRIPTION
**Pull request by aktech**
TODO
- [x] Return dictionary of linsolve solution 
- [x] https://github.com/sympy/sympy/pull/10000#discussion_r42305598
- [ ] Fix Failing Tests

@asmeurer @hargup

### CODE REVIEW

File: sympy/solvers/solvers.py [hunk 1]
```diff
@@ -2046,7 +2047,7 @@ def update(determined, solution):
             bestsol = thissol
         return bestsol
 
-
+@deprecated(useinstead="linsolve", issue=10001, deprecated_since_version="0.7.6")
```
Comment (asmeurer) [inline sympy/solvers/solvers.py]: This should be 0.7.7 (actually 1.0, but we'll have to change all instances of the former to the latter at some point anyway).

**Comment by asmeurer** [comment]
Nice, issue 10000!

**Comment by asmeurer** [comment]
The tests need to be fixed.

**Comment by asmeurer** [comment]
It's good to see this by the way. I think this fixes https://github.com/sympy/sympy/issues/4949.

**Comment by aktech** [comment]
>  I think this fixes #4949.

There are few more functions which duplicates solving linear equations, will try to finish this soon.

Here are all the permutations:
```
solve(equations, unknowns)
solve_linear_system(aug_matrix)
solve_linear_system_LU(aug_matrix, unknowns)
solve_poly_system(equations, unknowns)
solve_linear(single_equation)
polys.solvers.solve_lin_sys(equations, ring)
minsolve_linear_system(system, symbols)
```

File: sympy/solvers/solvers.py [hunk 2]
```diff
@@ -2087,162 +2088,8 @@ def solve_linear_system(system, *symbols, **flags):
...
-            else:
-                solutions[syms[k]] = content
-
-            k -= 1
-
-        return solutions
-    else:
-        return []   # no solutions
+    from sympy.solvers.solveset import linsolve
+    return linsolve(system, symbols)
```
Comment (asmeurer) [inline sympy/solvers/solvers.py]: Is the call signature different for linsolve (this function uses `*symbols`)? If so, you might mention that in the deprecation notice.

Comment (aktech) [inline sympy/solvers/solvers.py]: Indeed.

Comment (aktech) [inline sympy/solvers/solvers.py]: The `**flags` have been removed.

**Comment by aktech** [comment]
It seems there are some critical issues with `rref` which effects `linsolve`. If we have to release 1.0 soon, then we should skip deprecating this for now. We should just document that `solve_linear_system` will be replaced by `linsolve` and people should use linsolve. Similiar to what we would do with `solveset` (`linsolve` is a part of `solveset` module).

**Comment by oscarbenjamin** [comment]
This is somewhat out of date now.

**Pull request closed without merging**